njsscan/njsscan.py

```diff
@@ -22,9 +22,8 @@ def __init__(self, paths, json, check_controls) -> None:
             'ignore_extensions': conf['ignore_extensions'],
             'ignore_paths': conf['ignore_paths'],
             'ignore_rules': conf['ignore_rules'],
+            'show_progress': show_progress,
         }
-        if not json:
-            self.options['show_progress'] = True
```

Comment:
It was intentional to show progress bar only for njsscan's default CLI output format. This change will enable progress bar only for json and sonarqube output formats and disable it for njsscan CLI. 

Is there a particular reason for enabling progress bar for json and sonarqube output? I guess this output formats will be used in automated CI or programmatic use cases and make no sense to show a progress bar. However, users running njsscan in command-line interface should see the progress bar.